uxarray/grid.py

```diff
@@ -0,0 +1,279 @@
+"""uxarray grid module."""
+import os
+import xarray as xr
+import numpy as np
+from pathlib import PurePath
+
+
+class Grid:
+    """A class for uxarray grid object.
+
+    Examples
+    ========
+    import uxarray as ux
+    # open an exodus file with Grid object
+    mesh = ux.Grid("filename.g")
+
+    # save as ugrid file
+    mesh.saveas("outfile.ug")
+    """
+
+    # Import read/write methods from files in this folder
+    from ._exodus import read_exodus, write_exodus
+    from ._ugrid import read_ugrid, write_ugrid
+    from ._shpfile import read_shpfile
+    from ._scrip import read_scrip
+
+    def __init__(self, *args, **kwargs):
```

Comment:
From the whole implementation of this function, I think that taking arguments as `**kwargs` only (i.e. not `*args` in addition) could help us simplify the flow. For instance, if "input_file" in kwargs, then initialize from a file, else if "vertices" in kwargs, then initialize from vertices, etc.

Reply:
I would like to have this option via args:

`mesh = ux.Grid("filename.g")`

There is logic to automatically identify several of the parameters specified in kwargs. How about we remove kwargs?

Reply:
Yes, that makes to keep `args` after seeing the Grid constructor you gave. Not sure about removing `kwargs` as it may make the implementation more difficult. I think after you address some of the inline comments in this same function, we may have a better flow of the function.

Reply:
I would suggest using optionals set to `None` by default, rather than a `**kwargs` expecting a dictionary. 
it would be more 'pythonic' than the old 'c style' argments, and has the advantage of being cleaner to implement and far easier to maintain and document.